python/tutorials/09-persistent-matmul.py

```diff
@@ -541,41 +541,59 @@ def torch_matmul(a, b):
     return c
 
 
-def bench(K, dtype, tiles_per_update, reps=10):
+@contextmanager
+def proton_context():
+    proton.activate(0)
+    try:
+        yield
+    finally:
+        proton.deactivate(0)
+
+
+def bench(K, dtype, tiles_per_update, reps=1000, warmup_reps=10000):
     M = 8192
     N = 8192
     a = torch.randn((M, K), device="cuda", dtype=torch.float16).to(dtype)
     b = torch.randn((K, N), device="cuda", dtype=torch.float16).to(dtype)
 
     b = b.T.contiguous()
 
-    proton.activate(0)
-
     if cublas is not None:
-        for _ in range(reps):
+        for _ in range(warmup_reps):
```

Comment:
Maybe it could be a function that performs benchmarking?

```
    def bench(reps, warpup_reps, fn):
        for _ in range(warpup_reps):
            fn(a, b)
        with proton_context():
            for _ in range(reps):
                fn(a, b)
```

Reply:
This way you can reuse the function for different `fn`s